File: AccountManager.Business/Application/StatementService.cs
using AccountManager.Business.Api;
using AccountManager.Business.Api.Models;
using AccountManager.Business.Application.Mappers;
using AccountManager.Business.Domain.Infrastructure;

namespace AccountManager.Business.Application
{
    public class StatementService : IStatementService
    {
        private readonly IStatementItemRepository _statementItemRepository;
        public StatementService(IStatementItemRepository statementItemRepository)
        {
            _statementItemRepository = statementItemRepository;
        }

        public GetStatementItemsResponse GetStatementItems()
        {
            var statementItems = _statementItemRepository.GetStatementItems();

            return GetStatementItemsResponseMapper.Map(statementItems);
        }
    }
}

File: AccountManager.Node/app/web/services/statementService.ts
/// <reference path="../models/models.ts" />
import domain = require('../../domain/models');
import models = require('../models/models')
import api = require('../services/apiService')

export class StatementService {
    accountManagerModel: models.AccountManagerModel;
    private apiService: api.ApiService
    static $inject = ["accountManager.Services.ApiService", "accountManager.Models.accountManagerModel"];
    constructor(apiService: api.ApiService, accountManagerModel: models.AccountManagerModel) {
        this.apiService = apiService;
        this.accountManagerModel = accountManagerModel;
    }

    getStatementItems = () => {

        var result = this.apiService.getStatementItems();

        result.then(apiResult => {
            console.log("apiResult");
            console.log(apiResult.statementItems);
            this.accountManagerModel.statementItems = apiResult.statementItems;
        });

        return result;
    }

    getCategories(): angular.IPromise<domain.Category[]> {
        var result = this.apiService.getCategories();

        result.then(apiResult => {

            this.accountManagerModel.categories = apiResult.data.categories;
        });

        return result;
    }
}
angular.module("accountManager").service("accountManager.Services.StatementService", StatementService);